src/main/java/com/demo/MybatisApplication/service/StudentService.java

```diff
@@ -1,51 +1,55 @@
 package com.demo.MybatisApplication.service;
 
-import com.demo.MybatisApplication.dto.StudentAddDto;
-import com.demo.MybatisApplication.dto.StudentDisplayAsSubjects;
-import com.demo.MybatisApplication.dto.StudentDisplayByIdDto;
-import com.demo.MybatisApplication.dto.StudentsDisplayDto;
+import com.demo.MybatisApplication.dto.StudentAdditionDto;
+import com.demo.MybatisApplication.dto.StudentSubjectsDisplayDto;
+import com.demo.MybatisApplication.dto.StudentDisplayDto;
+import com.demo.MybatisApplication.dto.SubjectEntityDisplayDto;
+import com.demo.MybatisApplication.mapstruct.SubjectMapper;
+import com.demo.MybatisApplication.mapstruct.SubjectMapperImpl;
 import com.demo.MybatisApplication.model.StudentEntity;
 import com.demo.MybatisApplication.model.SubjectEntity;
+import com.demo.MybatisApplication.model.TeacherEntity;
 import com.demo.MybatisApplication.repository.StudentRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
 import com.demo.MybatisApplication.mapstruct.StudentMapper;
-import org.springframework.web.bind.annotation.RequestBody;
 
 import java.util.List;
+import java.util.stream.Collectors;
 
 @Service
 public class StudentService {
     @Autowired
     private StudentRepository studentRepository;
-
     @Autowired
     private StudentMapper studentMapper;
 
-    public StudentDisplayByIdDto getStudentById(Long id){
+    public StudentDisplayDto getStudentById(Long id){
         StudentEntity studentEntity = studentRepository.findStudentById(id);
-        return studentMapper.studentEntityToDisplayByIdDto(studentEntity);
+        return studentMapper.mapStudentEntityToDisplayByIdDto(studentEntity);
     }
 
-    public void assignSubjectsToStudent(Long studentId, List<SubjectEntity> subjects) {
-        studentRepository.updateSubjectsToStudent(studentId, subjects);
+    public void assignSubjectsToStudent(Long studentId, List<SubjectEntityDisplayDto> subjects) {
+        SubjectMapperImpl subjectMapper = new SubjectMapperImpl();
+        List<SubjectEntity> subjectEntities = subjects.stream()
```

Comment:
Create a method in mapper to map the list instead of looping over here.

Reply:
Done